.NET 6.0 can't be used anymore to build C# solutions in a project

### Tested versions

- Reproducible in: 4.3.dev f040a351c2f27c6b86c40c71a28babf99dd0b9bd


### System information

Godot v4.3.dev (f040a351c2f27c6b86c40c71a28babf99dd0b9bd) - Fedora Linux 39 (KDE Plasma) - Wayland - GLES3 (Compatibility) - NVIDIA GeForce RTX 4090 (nvidia; 545.29.06) - 13th Gen Intel(R) Core(TM) i9-13900K (32 Threads)

### Issue description

I've been trying to run https://github.com/godotengine/godot-benchmarks on the benchmarks server (which runs Fedora 39 just like my PC), but I'm getting an error:

```
  Determining projects to restore...
  Restored /home/godot/godot-benchmarks/godot/modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators.Sample/Godot.SourceGenerators.Sample.csproj (in 82 ms).
  Restored /home/godot/godot-benchmarks/godot/modules/mono/editor/Godot.NET.Sdk/Godot.NET.Sdk/Godot.NET.Sdk.csproj (in 119 ms).
  Restored /home/godot/godot-benchmarks/godot/modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators.Tests/Godot.SourceGenerators.Tests.csproj (in 5.04 sec).
  3 of 6 projects are up-to-date for restore.
  Godot.NET.Sdk ->
  Godot.SourceGenerators -> /home/godot/godot-benchmarks/godot/modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators/bin/Release/netstandard2.0/Godot.SourceGenerators.dll
  Successfully created package '/home/godot/godot-benchmarks/godot/modules/mono/editor/Godot.NET.Sdk/Godot.NET.Sdk/bin/Release/Godot.NET.Sdk.4.3.0-dev.nupkg'.
  Godot.SourceGenerators.Internal -> /home/godot/godot-benchmarks/godot/modules/mono/glue/GodotSharp/Godot.SourceGenerators.Internal/bin/Debug/netstandard2.0/Godot.SourceGenerators.Internal.dll
  GodotSharp -> /home/godot/godot-benchmarks/godot/modules/mono/glue/GodotSharp/GodotSharp/bin/Release/GodotSharp.dll
CSC : error CS1617: Invalid option '11' for /langversion. Use '/langversion:?' to list supported values. [/home/godot/godot-benchmarks/godot/modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators.Sample/Godot.SourceGenerators.Sample.csproj]
CSC : error CS1617: Invalid option '11' for /langversion. Use '/langversion:?' to list supported values. [/home/godot/godot-benchmarks/godot/modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators.Tests/Godot.SourceGenerators.Tests.csproj]
```

I have `dotnet-sdk-6.0` installed from the Fedora repositories on the server. On my system, I also have `dotnet-sdk-7.0` and `dotnet-sdk-8.0` installed and it works.

As far as I know, we still try to support .NET 6.0 as a baseline, so it should be possible to build solutions for desktop platforms without having to install a newer version.

According to @paulloz, this may be related to https://github.com/godotengine/godot/pull/86528. cc @avilches

### Steps to reproduce

```bash
git clone --depth=1 https://github.com/godotengine/godot-benchmarks.git
cd godot-benchmarks
path/to/godot.binary --editor --build-solutions --quit-after 100
```

### Minimal reproduction project (MRP)

https://github.com/godotengine/godot-benchmarks

--- a/modules/mono/build_scripts/build_assemblies.py
+++ b/modules/mono/build_scripts/build_assemblies.py
@@ -151,7 +151,7 @@ def find_any_msbuild_tool(mono_prefix):
     return None
 
 
-def run_msbuild(tools: ToolsLocation, sln: str, msbuild_args: Optional[List[str]] = None):
+def run_msbuild(tools: ToolsLocation, sln: str, chdir_to: str, msbuild_args: Optional[List[str]] = None):
     using_msbuild_mono = False
 
     # Preference order: dotnet CLI > Standalone MSBuild > Mono's MSBuild
@@ -190,7 +190,8 @@ def run_msbuild(tools: ToolsLocation, sln: str, msbuild_args: Optional[List[str]
             }
         )
 
-    return subprocess.call(args, env=msbuild_env)
+    # We want to control cwd when running msbuild, because that's where the search for global.json begins.
+    return subprocess.call(args, env=msbuild_env, cwd=chdir_to)
 
 
 def build_godot_api(msbuild_tool, module_dir, output_dir, push_nupkgs_local, precision):
@@ -218,11 +219,7 @@ def build_godot_api(msbuild_tool, module_dir, output_dir, push_nupkgs_local, pre
             args += ["/p:GodotFloat64=true"]
 
         sln = os.path.join(module_dir, "glue/GodotSharp/GodotSharp.sln")
-        exit_code = run_msbuild(
-            msbuild_tool,
-            sln=sln,
-            msbuild_args=args,
-        )
+        exit_code = run_msbuild(msbuild_tool, sln=sln, chdir_to=module_dir, msbuild_args=args)
         if exit_code != 0:
             return exit_code
 
@@ -361,7 +358,7 @@ def build_all(msbuild_tool, module_dir, output_dir, godot_platform, dev_debug, p
         args += ["/p:ClearNuGetLocalCache=true", "/p:PushNuGetToLocalSource=" + push_nupkgs_local]
     if precision == "double":
         args += ["/p:GodotFloat64=true"]
-    exit_code = run_msbuild(msbuild_tool, sln=sln, msbuild_args=args)
+    exit_code = run_msbuild(msbuild_tool, sln=sln, chdir_to=module_dir, msbuild_args=args)
     if exit_code != 0:
         return exit_code
 
@@ -372,7 +369,7 @@ def build_all(msbuild_tool, module_dir, output_dir, godot_platform, dev_debug, p
     if precision == "double":
         args += ["/p:GodotFloat64=true"]
     sln = os.path.join(module_dir, "editor/Godot.NET.Sdk/Godot.NET.Sdk.sln")
-    exit_code = run_msbuild(msbuild_tool, sln=sln, msbuild_args=args)
+    exit_code = run_msbuild(msbuild_tool, sln=sln, chdir_to=module_dir, msbuild_args=args)
     if exit_code != 0:
         return exit_code
 

--- a/modules/mono/global.json
+++ b/modules/mono/global.json
@@ -1,0 +1,6 @@
+{
+  "sdk": {
+    "version": "8.0.0",
+    "rollForward": "latestMajor"
+  }
+}

